src/util/simplify_expr.cpp

```diff
@@ -1373,7 +1373,7 @@ bool simplify_exprt::simplify_update(exprt &expr)
       std::size_t number=to_struct_type(value_ptr_type).
         component_number(component_name);
 
-      assert(number<value_ptr->operands().size());
+      CHECK_RETURN(number < value_ptr->operands().size());
```

Comment:
This code should be rewritten to use `get_component`.

Reply:
Refactored.